velocity/src/main/java/net/skinsrestorer/velocity/SkinsRestorerVelocity.java

```diff
@@ -204,12 +173,17 @@ public InputStream getResource(String resource) {
 
     @Override
     public void runAsync(Runnable runnable) {
-        proxy.getScheduler().buildTask(this, runnable).schedule();
+        proxy.getScheduler().buildTask(pluginInstance, runnable).schedule();
+    }
+
+    @Override
+    public void runSync(Runnable runnable) {
+        runAsync(runnable);
     }
```

Comment:
Can you explain this?

Reply:
Some platforms don't run on a main thread, so I made this fallback method call instead of throwing a exception. Most API can be called async on those platforms.

Reply:
But why is the method called runsync and it does async?

Reply:
It's a method that calls scheduleSxncDelayedTask on other platformd to be in sync with the main thread. Other platforms like bungee and velocity don't have a main thread scheduler, that's why I'm defaulting to async there.